net/query.go

```diff
@@ -54,22 +53,16 @@ func (h *host) Query(who peer.AddrInfo) (*QueryResponse, error) {
 
 func (h *host) receiveQueryResponse(stream libp2pnetwork.Stream) (*QueryResponse, error) {
 	h.queryMu.Lock()
-	defer h.queryMu.Unlock()
-
-	buf := h.queryBuf
+	defer h.queryMu.Unlock() // TODO: Do we need this?
```

Comment:
since you removed the buffer then now, but I think we should have the pre-allocated buffer